includes/blocks.inc

```diff
@@ -888,3 +896,98 @@ function islandora_solr_sort() {
 
   return $output;
 }
+
+/**
+ * Search results navigation block.
+ *
+ * Pull params out of the URL to determine the link to return
+ * the original search, and the previous and next object in the search
+ * according to the current object.
+ *
+ * @return string
+ *   The rendered html for the block content.
+ */
+function islandora_solr_search_navigation() {
+  $return = NULL;
+  $params = drupal_get_query_parameters();
+  if (variable_get('islandora_solr_search_navigation', FALSE) && isset($params['islandora_solr_search_nav_params'])) {
+    $prev_pid = NULL;
+    $next_pid = NULL;
+    $current_pid = (islandora_is_valid_pid(arg(2)) ? arg(2) : NULL);
+    $current_pid_key = NULL;
+    $params = $params['islandora_solr_search_nav_params'];
+    $page = (isset($params['solr_page']) ? $params['solr_page'] : 0);
+    $query_processor = new IslandoraSolrQueryProcessor();
+    $query_processor->buildQuery($params['solr_query'], $params['solr_internal_params']);
+    $limit = $query_processor->solrLimit;
+    $query_processor->solrStart = $page * $limit;
+    $query_processor->executeQuery();
+
+    if (!empty($query_processor->islandoraSolrResult)) {
+      $objects = $query_processor->islandoraSolrResult['response']['objects'];
+      $limit = $query_processor->solrLimit;
+      $num_found = $query_processor->islandoraSolrResult['response']['numFound'];
+      if ($num_found > 1 && $current_pid) {
+        foreach ($objects as $key => $object) {
+          if ($object['PID'] == $current_pid) {
+            $current_pid_key = $key;
+            break;
+          }
+        }
+        if ($current_pid_key !== NULL) {
```

Comment:
There is an issue here. As the Solr results are alterable modules such as the Book SP, https://github.com/Islandora/islandora_solution_pack_book/blob/7.x/islandora_book.module#L199-L236, we can't rely on the PID being grabbed from the URL to be equivalent to the search result.

That being said, you could probably rely upon the offset and store it with your other parameters.